### DISCUSSION TITLE
When will the new version be the released?

### DISCUSSION THREAD

**Discussion by alexmamonchik** [stripe-ruby-mock/stripe-ruby-mock#887] [Announcements]
This discussion has the same topic as #795. Looks like people still use StripeRubyMock in 2023, but we know that it's too outdated. There is no just one active maintainer. All of us open Pull Requests because we need something in our projects. I would like to spend some time and resuscitate StripeRubyMock.

Problems that I see now:

1. current code supports old Stripe API versions

I don't think we have to support all versions. I like the [idea](https://github.com/stripe-ruby-mock/stripe-ruby-mock/issues/795#issuecomment-1697244317) of @TastyPi  to release v4.0 and try to use the newest Stripe API only.

_What about devs who use old Stripe API version and ruby 2.4?_

If need we can use another branch for old API versions, but we are not going to improve it. I hope it can be another small reason to upgrade technologies.

2. Test coverage. RIP Travis. I already added Github Actions #886 - feel free to improve and extend it. For Now I've added ruby 2.6, 2.7, 3.0. 

I think in v4.0 we'll use 2.7, 3.0, 3.2. Any suggestions?

I would like to see in Actions [SimpleCov](https://github.com/simplecov-ruby/simplecov) and [Rubocop](https://github.com/rubocop/rubocop). 

3. Too much opened Issues and Pull Requests.

Some of them are really good and should be merged before v4.0. I'll focus on that right now. I'm looking at PRs and add the label "will be in v4". I'll try to merge another PRs. I think soon we can release 3.1.

4. Gitter. Not a big deal, but I think Discussions will be better. Let's try.

Please, share your thoughts how to make StripeRubyMock great again here.

**Comment by sgerrand** [comment]
Thanks for sharing your thoughts on how to modernise this gem and make it usable again for contemporary use. I don't have anything to add to what you've suggested but I endorse them all as improvements and will be happy to help make those changes.

**Comment by alexmamonchik** [comment]
I'm not sure what should be included in v4: only stripe v10 or create own major version for each breaking changes, for example stripe gem v7, v8, v9? Maybe own branch for every version? When somebody wants to add a fix for stripe v7 - PR will be merged to specific branch

**Reply by TastyPi** [reply]
I would suggest doing whatever is easiest to maintain, even if that means only supporting the latest version of the Stripe API. The harder this project is to maintain, the less likely it will be going forward. If there is enough demand to support older versions then it can be considered, ideally with sponsorship/contributions from the companies that need it.

**Comment by chriskottom** [comment]
Two questions:

- What can we do to help?
- Will there be a v4 working branch that we can start testing on soon?

**Reply by alexmamonchik** [reply]
we have a lot of opened pull requests. Most of them are already outdated and should be closed. Another should be merged now - they will be useful for v4 too. Will be great if you can help with review existing PRs. Then we'll create another v3 release and create another branch for v3 only. Master branch will be used to v4 only and added all newest versions of gems and api.

**Comment by alexmamonchik** [comment]
We've finally released v3.1.0. Then we can support the newest Stripe version only. All next changes should use only newest Stripe version.

**Reply by jacobjlevine** [reply]
Fully endorse this. Good luck with continued testing. Can't wait to use v4 once it's ready. Any sense of timeline for that?

**Reply by alexmamonchik** [reply]
there is no date. But you can help with review opened pull requests. Start with features you need. I'm trying to get first version of updated api https://github.com/stripe-ruby-mock/stripe-ruby-mock/pull/897

**Comment by alexmamonchik** [comment]
hi. I've just released v4.0.0. Now I hope we can update this gem to the newest Stripe API (so close as we need for our projects).I think we have to remove deprecated API.

**Reply by yunixon** [reply]
Version 4 still not working for me

[Pay] stripe gem must be version ~> 12
Could not find compatible versions

Because every version of stripe-ruby-mock depends on stripe > 5, < 12
  and Gemfile depends on stripe ~> 12
